### PR TITLE
Updated data tutorials

### PR DESCRIPTION
**Pull request by thomelane**
Simplified data tutorial to make more beginner friendly. This tutorial introduces the main components of the data pipeline (i.e. Dataset, tranforms and DataLaoders) and shows how they fit together. Will follow up with seperate and dedicated tutorials on each going into more of the details and advanced concepts.

### CODE REVIEW

**Comment by thomelane** [comment]
@Ishitori please review, thanks!

File: python/guide/packages/gluon/data.md [hunk 1]
```diff
@@ -1,325 +1,120 @@
...
-import tarfile
-import time
-
-from mxnet import nd, image, io
-from mxnet.gluon import utils, data as gdata
-```
+One of the most critical steps for model training and inference is loading the data: without data you can’t do deep learning! In this tutorial we use the `data` module to:
 
-## Load and Parse Datasets
+1) define a [`Dataset`](https://beta.mxnet.io/api/gluon/_autogen/mxnet.gluon.data.Dataset.html)
```
Comment (Ishitori) [inline python/guide/packages/gluon/data.md]: I think upper-case looks better in the list:
1) Define ...
2) Use ...
3) Use ...

File: python/guide/packages/gluon/data.md [hunk 2]
```diff
@@ -1,325 +1,120 @@
...
-necessary when these transformations contain randomness. But we can also apply
-all transformations during creating the dataset.
-
-```{.python .input}
-dogcat_cached = dogcat.transform_first(resize, lazy=False)
+transform_fn = RandomResizedCrop(size=(20, 20), scale=(0.6, 0.8))
 ```
 
-So no transform is needed during accessing examples.
+After defining `transform_fn`, we now need to apply it to the dataset. We call the `transform_first` method of our dataset to apply `transform_fn` to the first element of all samples. We had 2 elements per sample in our example, so we only apply `transform_fn` to the image and not the label.
```
Comment (Ishitori) [inline python/guide/packages/gluon/data.md]: I would add a short sentence on the fact, that it is a lazy transformation.

File: python/guide/packages/gluon/data.md [hunk 3]
```diff
@@ -1,325 +1,120 @@
...
 
-```{.python .input}
-dogcat_cached[0][0].shape
+```
+dataset = dataset.transform_first(transform_fn)
 ```
 
-Besides `transform_first`, we can apply transform to all entries in an example.
-The following examples add the label by 10 besides resizing an image:
+When we retrieve the same sample as before from the dataset, we now see an augmented version of the image (with the same label). We'd see a different augmented image every time we retrieve this sample.
```
Comment (Ishitori) [inline python/guide/packages/gluon/data.md]: I would add a sentence, that if you want to convert all elements, then use transform() function instead of transform_first.

File: python/guide/packages/gluon/data.md [hunk 4]
```diff
@@ -1,325 +1,120 @@
...
-Or put it into the beginning of the next epoch:
-
-```{.python .input  n=23}
-data = gdata.DataLoader(dataset, batch_size=4, last_batch='rollover')
-for X, y in data:
-    print('epoch 0, y:', y.asnumpy())
-for X, y in data:
-    print('epoch 1, y:', y.asnumpy())
+dataloader = mx.gluon.data.DataLoader(dataset,
+                                      batch_size=12345,
```
Comment (Ishitori) [inline python/guide/packages/gluon/data.md]: Let's try giving meaningful numbers, like 100, in case it would fail with OOM on a small machine.

File: python/guide/packages/gluon/data.md [hunk 5]
```diff
@@ -1,325 +1,120 @@
...
 
-```{.python .input}
-data_iter = io.NDArrayIter(data=features, label=labels, batch_size=4)
-data = DataIterLoader(data_iter)
-for X, y in data:
-    print('X shape:', X.shape, '\ty:', y.asnumpy())
-```
+## Recommended Next Steps
+
+We used presets and a lot of default values in this tutorial, but there are many other options and possibilities. You're likely to want to use your own dataset for training. You can learn more about this in the dedicated `Dataset` tutorial. We also have a tutorial on `transform`s with examples on how to compose transform operations. And last but not least, we have a very useful tutorial on `DataLoader`s. You'll often find that loading data can be the bottleneck to training, and this tutorial contains some useful tricks to speed up the data pipeline.
```
Comment (Ishitori) [inline python/guide/packages/gluon/data.md]: We mention 3 extra tutorials here: transform, Dataset and Dataloader. Can we provide links to them here?

Comment (thomelane) [reply python/guide/packages/gluon/data.md]: We don't have these right now, but will add links when they're done.

File: python/guide/packages/gluon/data.md (hunk 4 above)
Comment (thomelane) [reply python/guide/packages/gluon/data.md]: Was trying to avoid too many batches showing. Will add a comment about this though.

File: python/guide/packages/gluon/data.md (hunk 3 above)
Comment (thomelane) [reply python/guide/packages/gluon/data.md]: Done.

File: python/guide/packages/gluon/data.md (hunk 2 above)
Comment (thomelane) [reply python/guide/packages/gluon/data.md]: Done, a bit further down.

File: python/guide/packages/gluon/data.md (hunk 1 above)
Comment (thomelane) [reply python/guide/packages/gluon/data.md]: Done.

**Comment by thomelane** [comment]
thanks for the useful feedback @Ishitori!
